support/cas-server-support-pac4j-core-clients/src/main/java/org/apereo/cas/support/pac4j/web/flow/DelegatedClientAuthenticationAction.java

```diff
@@ -331,6 +340,31 @@ private boolean isDelegatedClientAuthorizedForService(final Client client, final
         return true;
     }
 
+    /**
+     * Prepares for a future SAML Single Logout by saving the current user profile through a PAC4J profile service.
+     * 
+     * @param client
+     *            The current PAC4J client.
+     * @param credentials
+     *            Credentials from the user.
+     * @param webContext
+     *            PAC4J web context.
+     * @param tgtId
+     *            TGT ID, will be used as the Linked ID for later retrieval.
+     * 
+     * @throws Exception
+     *             If anything fails.
+     */
+    private void prepareForFutureSingleLogout(final BaseClient<Credentials, ? extends UserProfile> client, final Credentials credentials,
+            final WebContext webContext, final String tgtId) throws HttpAction {
+        if (profileService != null) {
```

Comment:
Can this ever be null? and if so, we should assert to make sure it's not or throw an error?

Reply:
It's set in the constructor, like other attributes, so in theory it could be null. I will add an Assert.notNull() check to the constructor and remove this if(). Just notice the other attributes are not checked either, so maybe we should add more checks to keep a consistent approach to all.

Reply:
Finally I added checks for all constructor arguments, as all attributes should be non-null. And I adapted the unit test little bit.